src/test/kotlin/de/danielmaile/mpp/item/ItemRegistryTest.kt

```diff
@@ -0,0 +1,137 @@
+        }
+        assertTrue(containsAll)
+    }
+
+    @Test
+    fun `Registering armors should register all armors`() {
+        ItemRegistry.registerItems(Armors.values())
+        var containsAll = true
+        for (item in Armors.values()) {
+            if (!ItemRegistry.getAllItems().contains(item)) containsAll = false
+        }
+        assertTrue(containsAll)
+    }
+
+    @Test
+    fun `Registering ingredients should register all ingredients`() {
+        ItemRegistry.registerItems(Ingredients.values())
+        var containsAll = true
+        for (item in Ingredients.values()) {
+            if (!ItemRegistry.getAllItems().contains(item)) containsAll = false
+        }
+        assertTrue(containsAll)
+    }
+
+    @Test
+    fun `Registering tools should register all tools`() {
+        ItemRegistry.registerItems(Tools.values())
+        var containsAll = true
+        for (item in Tools.values()) {
+            if (!ItemRegistry.getAllItems().contains(item)) containsAll = false
+        }
+        assertTrue(containsAll)
+    }
+
+    @Test
+    fun `There should not be any repeated model ID`() {
+        registerItems()
+        val modelIDs = mutableListOf<Int>()
+        var repeats = false
+        for (item in ItemRegistry.getAllItems()) {
+            if (modelIDs.contains(item.modelID)) {
+                repeats = true
+                break
+            }
+            modelIDs.add(item.modelID)
+        }
+        assertFalse(repeats)
+    }
+
+    @Test
+    fun `Clearing should clear the whole registry`() {
+        ItemRegistry.registerItems(Blocks.values())
+        ItemRegistry.clear()
+        assertEquals(0, ItemRegistry.getAllItems().size)
+    }
+
+    @Test
+    fun `Registering all items should register all items`() {
+        registerItems()
+        val items = Blocks.values().size + Tools.values().size + Armors.values().size + Ingredients.values().size
+        assertEquals(items, ItemRegistry.getAllItems().size)
+    }
+
+    @Test
+    fun `Getting an unregistered item by id should throw a NoSuchElementException`() {
+        assertThrows<NoSuchElementException>() {
+            ItemRegistry.getItemByID("abc:idontexist")
+        }
+    }
+
+    @Test
+    fun `Getting a registered item by id should not throw a NoSuchElementException`() {
+        ItemRegistry.registerItem(Blocks.AETHER_STONE)
+        assertDoesNotThrow {
+            ItemRegistry.getItemByID("mpp:aether_stone")
+        }
+    }
+
+    @Test
+    fun `Getting an unregistered item ID should throw a NoSuchElementException`() {
+        assertThrows<NoSuchElementException>() {
+            ItemRegistry.getItemID(Blocks.AETHER_STONE)
+        }
+    }
+
+    @Test
+    fun `Registering a single Item should register the Item as is`() {
+        ItemRegistry.registerItem(Blocks.AETHER_STONE)
+        assertEquals("mpp:aether_stone", ItemRegistry.getItemID(Blocks.AETHER_STONE))
```

Comment:
Shouldnt we use the variable from the Registry class instead of hard coding mpp:? Maybe we can remove the whole var because we will never change the prefix?

Reply:
I was thinking that using hardcoded values is more strict but that's inflexible too. While it is also true that we are not going to ever change prefix probably, also, I believe we should use namespaced keys now that you bring it up to our attention.

Reply:
Take a look at my latest comment regarding NamespacedKeys